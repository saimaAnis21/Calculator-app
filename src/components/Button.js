import PropTypes from 'prop-types';
import './styles.css';

function Button(props) {
  const { name } = props;
  const handleClick = (buttonName) => props.clickHandler(buttonName);
  const getClassName = (operator) => {
    if (operator === '=' || operator === '+' || operator === '-' || operator === 'X' || operator === '/') {
      return 'orng-btns w-25p h-100p brdr';
    } if (operator === '0') {
      return 'bg-btnpanel w-50p h-100p brdr';
    }
    return 'bg-btnpanel w-25p h-100p brdr';
  };

  return (
    <button type="submit" onClick={handleClick.bind(this, name)} className={getClassName(name)}>
      {name}
    </button>
  );
}

Button.propTypes = {
  name: PropTypes.string.isRequired,
  clickHandler: PropTypes.isRequired,
};

export default Button;
