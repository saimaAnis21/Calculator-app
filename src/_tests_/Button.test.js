import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Button from '../components/Button';
import '@testing-library/jest-dom';

describe('ButtonPanel renders correctly', () => {
  const test = () => { };

  it('renders correctly', () => {
    const tree = renderer
      .create(<Button name="AC" clickHandler={test} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with provided name', () => {
    render(<Button name="AC" clickHandler={test} />);
    expect(screen.getByText('AC')).toHaveTextContent('AC');
  });
});
