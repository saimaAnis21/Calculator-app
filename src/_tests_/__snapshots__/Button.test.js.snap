// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPanel renders correctly renders correctly 1`] = `
<button
  className="bg-btnpanel w-25p h-100p brdr"
  onClick={[Function]}
  type="submit"
>
  AC
</button>
`;
